### PR TITLE
Use theme stored in realm

### PR DESCRIPTION
**Pull request by tevincent**
https://github.com/user-attachments/assets/ddd2959b-204e-4eda-a3a8-d81057e0b426

### CODE REVIEW

File: app/src/main/java/com/infomaniak/swisstransfer/ui/theme/Theme.kt [hunk 1]
```diff
@@ -47,10 +52,11 @@ fun SwissTransferTheme(
...
 @Composable
-fun isDarkTheme(): Boolean {
-    // rememberMutableStateOf
-    // TODO check in realm. If system, isSystemDark, otherwise,
-    return isSystemInDarkTheme()
+fun isDarkTheme(settingsViewModel: SettingsViewModel): Boolean {
+    val appSettings by settingsViewModel.appSettingsFlow.collectAsStateWithLifecycle(null)
+    return appSettings?.let {
+        if (it.theme == Theme.SYSTEM) isSystemInDarkTheme() else it.theme == Theme.DARK
+    } ?: isSystemInDarkTheme()
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/theme/Theme.kt:59]: Is `System` in default from kmp?

**Review by sirambd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/ui/theme/Theme.kt (hunk 1 above)
Comment (tevincent) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/theme/Theme.kt:59]: Yes it is.

Capture d’écran 2024-09-19 à 09 28 53

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/settings/SettingsScreenWrapper.kt [hunk 2]
```diff
@@ -27,49 +27,83 @@ import androidx.compose.material3.adaptive.WindowAdaptiveInfo
...
+) {
     val context = LocalContext.current
     val aboutURL = stringResource(R.string.urlAbout)
     val userReportURL = stringResource(R.string.urlUserReportAndroid)
 
     SettingsScreen(
-        appSettings,
+        theme,
+        validityPeriod,
+        downloadLimit, emailLanguage,
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/settings/SettingsScreenWrapper.kt:106]:
```suggestion
        downloadLimit,
        emailLanguage,
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by KevinBoulongne — APPROVED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/settings/SettingsScreenWrapper.kt [hunk 3]
```diff
@@ -27,49 +27,84 @@ import androidx.compose.material3.adaptive.WindowAdaptiveInfo
...
+        listPane = { ListPane(navigator = this, theme, validityPeriod, downloadLimit, emailLanguage) },
+        detailPane = { DetailPane(navigator = this, theme, validityPeriod, downloadLimit, emailLanguage) },
+    )
+}
+
+@Immutable
+class GetSetCallbacks<T>(
+    val get: () -> T,
+    val set: (T) -> Unit,
+)
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/settings/SettingsScreenWrapper.kt:88]: Extract it into its own file

File: app/src/main/java/com/infomaniak/swisstransfer/ui/MainActivity.kt
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/MainActivity.kt:1]: We need to make the changes we've been talking about

**Review by sirambd — CHANGES_REQUESTED** [review]

**Review by sirambd — APPROVED** [review]

**Pull request merged**